src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -84,42 +83,45 @@ def dump_sources(builder, tgt, log):
                     'Depend on resources() targets instead.'.format(tgt.address.spec))
 
 
-def dump_requirements(builder, interpreter, req_libs, log, platforms=None):
+def dump_requirement_libs(builder, interpreter, req_libs, log, platforms=None):
   """Multi-platform dependency resolution for PEX files.
 
-  Returns a list of distributions that must be included in order to satisfy a set of requirements.
-  That may involve distributions for multiple platforms.
-
   :param builder: Dump the requirements into this builder.
   :param interpreter: The :class:`PythonInterpreter` to resolve requirements for.
   :param req_libs: A list of :class:`PythonRequirementLibrary` targets to resolve.
   :param log: Use this logger.
   :param platforms: A list of :class:`Platform`s to resolve requirements for.
                     Defaults to the platforms specified by PythonSetup.
   """
+  reqs = [req for req_lib in req_libs for req in req_lib.requirements]
+  dump_requirements(builder, interpreter, reqs, log, platforms)
+
+
+def dump_requirements(builder, interpreter, reqs, log, platforms=None):
+  """Multi-platform dependency resolution for PEX files.
+
+  :param builder: Dump the requirements into this builder.
+  :param interpreter: The :class:`PythonInterpreter` to resolve requirements for.
+  :param reqs: A list of :class:`PythonRequirement` to resolve.
+  :param log: Use this logger.
+  :param platforms: A list of :class:`Platform`s to resolve requirements for.
+                    Defaults to the platforms specified by PythonSetup.
+  """
 
   # Gather and de-dup all requirements.
-  reqs = OrderedSet()
-  for req_lib in req_libs:
-    for req in req_lib.requirements:
-      reqs.add(req)
+  deduped_reqs = OrderedSet()
```

Comment:
`deduped_reqs = OrderedSet(reqs)` should work.

Reply:
Ah yes, good call. Done!